### PR TITLE
Custom cell renderer for Modus Table

### PR DESCRIPTION
**Pull request by Ankur-Koul**
## Description

Added 2 types of Column types
- Link
- Badge

References #1306 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-data-table/modus-data-table.tsx [hunk 1]
```diff
@@ -90,6 +91,9 @@ export class ModusDataTable {
   /** Emits event on sort change */
   @Event() sortChange: EventEmitter<ModusDataTableSortingState>;
 
+  /** Emits event on sort change */
```
Comment (sankarNous) [inline stencil-workspace/src/components/modus-data-table/modus-data-table.tsx:94]: give a proper comment

File: stencil-workspace/src/components/modus-data-table/modus-data-table.tsx [hunk 2]
```diff
@@ -243,7 +247,15 @@ export class ModusDataTable {
               return (
                 <tr key={row.id} class={this.hover && 'enable-hover'}>
                   {row.getAllCells()?.map((cell) => {
-                    return <ModusDataTableCell cell={cell} />;
+                    return (
+                      <ModusDataTableCell
+                        cell={cell}
+                        onLinkClick={(link: ModusDataTableLink) => {
+                          this.linkClick.emit(link);
+                          console.log(link);
```
Comment (sankarNous) [inline stencil-workspace/src/components/modus-data-table/modus-data-table.tsx:255]: remove console

File: stencil-workspace/src/components/modus-data-table/models/modus-data-table-badge.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { ModusDataTableBadgeType } from '../enums';
+import { ModusDataTableBadgeColor } from '../enums/modus-data-table-badge-color';
+
+export interface ModusDataTableBadge {
+  color?: ModusDataTableBadgeColor;
+  text: string;
+  type?: ModusDataTableBadgeType;
+  _type: 'badge';
```
Comment (sankarNous) [inline stencil-workspace/src/components/modus-data-table/models/modus-data-table-badge.ts:8]: '_type'  should we name like this?

File: stencil-workspace/src/components/modus-data-table/models/modus-data-table-column.ts [hunk 4]
```diff
@@ -7,7 +7,8 @@ export interface ModusDataTableColumn<TData extends RowData, TValue = unknown> {
   dataType: ModusColumnDataType;
   id?: string;
   // columns?: ModusDataTableColumn[]; //To support in the future
-  cell?: ColumnDefTemplate<CellContext<TData, TValue>>;
+  cell?:
```
Comment (sankarNous) [inline stencil-workspace/src/components/modus-data-table/models/modus-data-table-column.ts:10]: it doesn't look right

File: stencil-workspace/src/components/modus-data-table/modus-data-table.tsx (hunk 1 above)
Comment (Ankur-Koul) [reply stencil-workspace/src/components/modus-data-table/modus-data-table.tsx]: Done

File: stencil-workspace/src/components/modus-data-table/readme.md [hunk 5]
```diff
@@ -24,9 +24,10 @@
 
 ## Events
 
-| Event        | Description                | Type                        |
-| ------------ | -------------------------- | --------------------------- |
-| `sortChange` | Emits event on sort change | `CustomEvent<ColumnSort[]>` |
+| Event        | Description                | Type                              |
+| ------------ | -------------------------- | --------------------------------- |
+| `linkClick`  | Emits event on sort change | `CustomEvent<ModusDataTableLink>` |
```
[the remark this answers is not in the document]
Comment (Ankur-Koul) [reply stencil-workspace/src/components/modus-data-table/readme.md]: Done

File: stencil-workspace/src/components/modus-data-table/models/modus-data-table-badge.ts (hunk 3 above)
Comment (Ankur-Koul) [reply stencil-workspace/src/components/modus-data-table/models/modus-data-table-badge.ts:8]: This is to override the type

File: stencil-workspace/src/components/modus-data-table/models/modus-data-table-column.ts (hunk 4 above)
Comment (Ankur-Koul) [reply stencil-workspace/src/components/modus-data-table/models/modus-data-table-column.ts]: Can you please elaborate on this?

File: stencil-workspace/src/components/modus-data-table/parts/cells/modus-data-table-cell.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import {
+  FunctionalComponent,
+  h, // eslint-disable-line @typescript-eslint/no-unused-vars
+} from '@stencil/core';
+import { Cell } from '@tanstack/table-core';
+import { ModusColumnDataType } from '../../enums/modus-column-data-type';
+import { PropertyDataType } from '../../constants/constants';
+import { cellFormatter } from './modus-data-table-cell-formatter';
```
Comment (sankarNous) [inline stencil-workspace/src/components/modus-data-table/parts/cells/modus-data-table-cell.tsx:8]: I think all exported functions/constants should be in PascalCase

File: stencil-workspace/storybook/stories/components/modus-data-table/modus-data-table.stories.tsx [hunk 7]
```diff
@@ -187,6 +187,32 @@ const valueFormatterTable = () => {
...
 
+export const CellRenderer = () => html`
+  <div style="width: 950px">
+    <modus-data-table
+      hover="true"
+      sort="true"
+      column-resize="false"
+      pagination="false"
+      show-sort-icon-hover="true"
+      summary-row="false" />
```
Comment (sankarNous) [inline stencil-workspace/storybook/stories/components/modus-data-table/modus-data-table.stories.tsx:198]: Can you make this values dynamic so we can control from this controls tab. I have already implemented and deployed in `dev/data-table-main`.

File: stencil-workspace/src/components/modus-data-table/enums/modus-column-data-type.ts [hunk 8]
```diff
@@ -1,6 +1,9 @@
 export enum ModusColumnDataType {
```
Comment (sankarNous) [inline stencil-workspace/src/components/modus-data-table/enums/modus-column-data-type.ts:1]: I think name should be like `ModusDataTableColumnDataType`
@msankaran0712 is that right?

File: stencil-workspace/src/components/modus-data-table/models/modus-data-table-column.ts (hunk 4 above)
Comment (sankarNous) [reply stencil-workspace/src/components/modus-data-table/models/modus-data-table-column.ts]: 'cell' type was removed. I think you have already undone this, as I could not find this change.

File: stencil-workspace/src/components/modus-data-table/parts/cells/modus-data-table-cell.tsx (hunk 6 above)
Comment (Ankur-Koul) [reply stencil-workspace/src/components/modus-data-table/parts/cells/modus-data-table-cell.tsx]: Not functions, this is true for interface, class and constants.

File: stencil-workspace/storybook/stories/components/modus-data-table/modus-data-table.stories.tsx (hunk 7 above)
Comment (Ankur-Koul) [reply stencil-workspace/storybook/stories/components/modus-data-table/modus-data-table.stories.tsx]: Done, please check.

**Pull request closed without merging**